assets/src/hooks/useNotificationsReducer.ts

```diff
@@ -183,9 +195,82 @@ const persistToggledNotificationReadState = (
   ])
 }
 
-export const useNotificationsReducer = (): [State, Dispatch] => {
+type InitialNotificationData = { initial_notifications: NotificationData[] }
+
+type InitialNotifications = { type: "initial"; payload: Notification[] }
+type NewNotification = { type: "new"; payload: Notification }
+
+type ReceivedNotifications = NewNotification | InitialNotifications | null
+
+const parseNotifications = (
+  notificationData: NotificationData | InitialNotificationData
+): NewNotification | InitialNotifications => {
+  if ("initial_notifications" in notificationData) {
+    return {
+      type: "initial",
+      payload: (
+        notificationData as InitialNotificationData
+      ).initial_notifications.map(notificationFromData),
+    }
+  }
+  return {
+    type: "new",
+    payload: notificationFromData(notificationData as NotificationData),
+  }
+}
+
+const useNotifications = (socket: Socket | undefined, routeIds: RouteId[]) => {
+  return useChannel<ReceivedNotifications>({
+    socket,
+    topic: "notifications",
+    event: "notification",
+    parser: parseNotifications,
+    dependencies: routeIds,
```

Comment:
Okay this is the piece I'm not sure about. Originally `useNotifications` had a dependency on routeIds so that the channel is re-joined every time the open list of routeIds changes. I'm not sure if that is necessary though - my impression from[ this line](https://github.com/mbta/skate/blob/e17a6a19ad7ee2a4ce9294c3322d58411696fe04/lib/notifications/notification_server.ex#L184) in NotificationServer is that if they didn't re-join the channel, they would still get notifications for the new route_ids.

Reply:
Also if we don't need to rejoin we can undo the addition of the `dependencies` arg to `useChannel`.

Reply:
> I was held up on the fact that you wouldn't get old notifications for a route when you add it, but we wouldn't have delivered those notifications to that user to begin with.

Interesting, I hadn't thought about rejoining to get the batch of all unexpired notifications for that new route as `initial_notifications`. To confirm, is the intended behavior that adding a new route means you'll only get new notifications for that route?

Reply:
> To confirm, is the intended behavior that adding a new route means you'll only get new notifications for that route?

Correct, if you didn't have the route open at the time that the notification was generated you shouldn't get it.

Reply:
> To confirm, is the intended behavior that adding a new route means you'll only get new notifications for that route?

Correct, if you didn't have the route open at the time that the notification was generated you shouldn't get it.

Reply:
> To confirm, is the intended behavior that adding a new route means you'll only get new notifications for that route?

Correct, if you didn't have the route open at the time that the notification was generated you shouldn't get it.